portfolio/src/main/java/com/google/sps/servlets/LoginStatusServlet.java

```diff
@@ -13,10 +14,29 @@
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
     response.setContentType("text/html;");
     UserService userService = UserServiceFactory.getUserService();
-    if(userService.isUserLoggedIn()) {
-      response.getWriter().println("loggedIn");
-    }else{
-      response.getWriter().println("notLoggedIn");
+    if (userService.isUserLoggedIn()) {
+      String email = userService.getCurrentUser().getEmail();
+      String link = userService.createLogoutURL("/");
+      response.getWriter().println(new Gson().toJson(new User(true, link, email)));
+    } else {
+      String link = userService.createLoginURL("/");
+      response.getWriter().println(new Gson().toJson(new User(false, link)));
+    }
+  }
+
+  static class User {
```

Comment:
Making this class static makes it feel like `User` is meant to be used by other classes (a static inner class in Java allows others to instantiate the inner class without having an instance of the parent class, e.g. in this case by saying `LoginStatusServlet.User user = new LoginStatusServlet.User()`. Because the default is not static, adding the `static` keyword makes it look like a deliberate design choice.

If it's not the case, and `User` is meant to be used only as a data-holder or struct-like class, only to be passed to Gson for formatting, then I would change `static` by `private`, which will not require any changes in the code but will make it clear that the class is meant to be used internally.
Otherwise, if other classes should be able to instantiate `User`, use it and pass it around, then it makes more sense for `User` to be in a different file (design-wise it's not very clean to have it declared inside a servlet).